service/src/main/java/bio/terra/workspace/service/policy/TpsApiDispatch.java

```diff
@@ -93,13 +97,13 @@ public void createPao(UUID workspaceUuid, @Nullable TpsPolicyInputs policyInputs
   // workspaces have an associated policy attribute object. This function creates
   // an empty one if it does not exist.
   @Traced
-  public void createPaoIfNotExist(UUID workspaceId) {
-    Optional<TpsPaoGetResult> pao = getPaoIfExists(workspaceId);
+  public void createPaoIfNotExist(UUID objectId, TpsComponent component, TpsObjectType objectType) {
```

Comment:
Is this change needed?
An alternative would have been to add an overload that populates the component and type.

Reply:
I made this change because I want to populate TDR snapshot PAOs if they don't exist. I agree that this is not really the province of WSM. However, I think it is important enough to do this anyway so that the PAO exists and future policy changes to the snapshot can trickle down appropriately.